api/API_VERSIONING.md

```diff
@@ -82,57 +78,17 @@ implementations within a major version should set explicit values for these fiel
-  rollouts as well. This previous stable major version will be supported for exactly 1
-  year after the introduction of the new current stable major version, after which it will be
-  removed from the Envoy implementation.
-* Optionally, the next experimental alpha major version, e.g. v4alpha. This is a release candidate
-  for the next stable major version. This is only generated when the current stable major version
-  requires a breaking change at the next cycle, e.g. a deprecation or field rename. This release
-  candidate is mechanically generated via the
-  [protoxform](https://github.com/envoyproxy/envoy/tree/main/tools/protoxform) tool from the
-  current stable major version, making use of annotations such as `deprecated = true`. This is not a
-  human editable artifact.
-
-An example of how this might play out is that at the end of December in 2020, if a v4 major version
-is justified, we might freeze
-`envoy.config.bootstrap.v4alpha` and this package would then become the current stable major version
-`envoy.config.bootstrap.v4`. The `envoy.config.bootstrap.v3` package will become the previous stable
-major version and support for `envoy.config.bootstrap.v2` will be dropped from the Envoy
-implementation. Note that some transitively referenced package, e.g.
-`envoy.config.filter.network.foo.v2` may remain at version 2 during this release, if no changes were
-made to the referenced package. If no major version is justified at this point, the decision to cut
-v4 might occur at some point in 2021 or beyond, however v2 support will still be removed at the end
-of 2020.
-
-The implication of this API lifecycle and clock is that any deprecated feature in the Envoy API will
-retain implementation support for at least 1-2 years.
-
-We are currently working on a strategy to introduce minor versions
-(https://github.com/envoyproxy/envoy/issues/8416). This will bump the xDS API minor version on every
-deprecation and field introduction/modification. This will provide an opportunity for the control
-plane to condition on client and major/minor API version support. Currently under discussion, but
-not finalized will be the sunsetting of Envoy client support for deprecated features after a year
-of support within a major version. Please post to https://github.com/envoyproxy/envoy/issues/8416
-any thoughts around this.
+At one point, the Envoy project planned for regular major version updates to the xDS API in order to
+remove technical debt. At this point we recognize that Envoy and the larger xDS ecosystem (gRPC,
+etc.) is too widely used to make version bumps realistic. As such, for practical purposes, the v3
+API is the final major version of the API and will be supported forever. Deprecations will still
+occur as an end-user indication that there is a preferred way to configure a particular feature, but
+no field will ever be removed nor will Envoy ever remove the implementation for any deprecated
```

Comment:
Should we mention the wrong API behavior could be fixed and the old behavior will be deprecated along with runtime flag?

Reply:
I think "wrong" is a bit too broad a term. We'll fix implementation behaviors that don't match the documented spec under runtime guard, is this what is meant?

Reply:
yea, agree with "wrong" isn't accurate. Not sure whether we doc about runtime and API somewhere.  I thought the rule could be:

* If the API doesn't work(It doesn't have any meaningful behavior), then it is just a bug, we fix it directly.
* If the API works with meaningful behavior, but the behavior is different from what the API doc said(or API doc doesn't have a clear description of the behavior), then we fix it with a runtime flag.

Reply:
I think we cover the use of runtime guarding in Envoy elsewhere (cc @alyssawilk) but happy to eat verbiage here if you can suggest some? In some sense I think runtime guarding is orthogonal to this.

Reply:
I only found this doc https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#runtime-guarding, we can reference it. But it doesn't talk about runtime flag for API, maybe we can improve that later.

Reply:
add few words there https://github.com/envoyproxy/envoy/pull/24134 if we didn't have such doc somewhere else